safe-actor/src/main/scala/com/evolutiongaming/safeakka/actor/SafeActorRef.scala

```diff
@@ -53,7 +53,12 @@ object SafeActorRef {
     def unsafe = ref.unsafe
     def tell[C](msg: C, sender: Option[ActorRef])(implicit canRcv: CanRcv[C, A]) = canRcv match {
       case CanRcv.Identity        => ref.tell(f(msg.asInstanceOf[A]), sender)(CanRcv.identity)
-      case canRcv: CanRcv.Sys[C] => ref.tell(msg, sender)(canRcv)
+      //workaround for Scala 3.3.0 compiler quirk:
+      //  asInstanceOf added because of a compilation error on pattern matching,
+      //  it complained about non-exhaustive match but suggested cases were non-existing
+      //
+      //old code: case canRcv: CanRcv.Sys[C] => ref.tell(msg, sender)(canRcv)
+      case canRcv: CanRcv.Sys[?] => ref.tell(msg, sender)(canRcv.asInstanceOf[CanRcv.Sys[C]])
```

Comment:
in https://users.scala-lang.org/t/alternative-for-type-ascriptions-after-tuple-patterns-in-scala-3-3-0/9328 I found alternative:
```suggestion
      case canRcv: CanRcv.Sys[C@unchecked] => ref.tell(msg, sender)(canRcv)
```

seems that it works in other cases too

Reply:
Applied where it fits